### ISSUE TITLE
CloudAppendBlob.ExistsAsync throws System.Xml.XmlException: 'Root element is missing.'

### ISSUE THREAD

**Issue by manums** [Azure/azure-storage-net#847]
### Which service(blob, file, queue, table) does this issue concern?
blob

### Which version of the SDK was used?
9.3.3

### Which platform are you using? (ex: .NET Core 2.1)
.Net Standard 2.0

### What problem was encountered?
blob.ExistsAsync() throws Xml Root element missing exception instead of returning false for non existing blob. Blob type is append bob. But blob did not exist anyway and hence we were expecting blob.ExistsAsync() to return false.

### How can we reproduce the problem in the simplest way?
1 - Try to get AppendBlobReference to some non existing blob name.
2 - With blob reference to CloudAppendBlob call ExistsAsync

### Have you found a mitigation/solution?
Nope. Tried workaround to fetch attributes and to check error code, but even that resulted in same exception.

**Requesting to hear any solution soon as this is blocking one critical component development**

![image](https://user-images.githubusercontent.com/3592179/51730533-a88d6780-202c-11e9-8958-eba731bbc9fb.png)

**Comment by crackalak** [comment]
I came across the same problem and raised issue #835

**Comment by crackalak** [comment]
This shouldn't stop you using ExistsAsync() though as the exception is handled in the SDK

**Comment by manums** [comment]
@crackalak  yes it is non-blocking & it is handled exception. I will update issue details.

**Comment by Trinitron** [comment]
@manums Hey, the same issue (System.Private.Xml.dll!System.Xml.XmlTextReaderImpl.Throw) for SDK v10.
Microsoft.Azure.Storage.Blob, Version=10.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
Target Framework: .Net Framework v4.5
`System.Xml.XmlException
  HResult=0x80131940
  Message=Root element is missing.
  Source=System.Private.Xml
  StackTrace:
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
`
